src/window.c

```diff
@@ -5328,7 +5328,7 @@ win_enter_ext(win_T *wp, int flags)
     if (*p_spk == 'c')		// assume cursor position needs updating
 	changed_line_abv_curs();
     else
-	win_fix_cursor(TRUE);
+	win_fix_cursor(get_real_state() & MODE_NORMAL);
```

Comment:
Does the check need to be the same as the one in `win_fix_cursor()`?
```c
    // Ensure cursor is valid when not in normal mode or when resized.
    if (!(get_real_state() & (MODE_NORMAL|MODE_CMDLINE|MODE_TERMINAL)))
	win_fix_cursor(FALSE);
    else if (resize)
	win_fix_cursor(TRUE);
```

Reply:
I think the edge cases covered by checking for `MODE_CMDLINE/TERMINAL` only apply when the caller is `win_fix_cursor()`. But I'm not sure, I guess it wouldn't hurt.

Reply:
Judging by the failing tests I guess I was wrong ;)